### DISCUSSION TITLE
Executing V in C

### DISCUSSION THREAD

**Discussion by armincosta** [vlang/v#20972] [Questions and Answers]
Hello,

V is an fantastic language!!:) I looked at the documentation and I'm wandering if there is the possibility to compile and execute V code within C code, like in TCC. 

Is there a convenient way to do this?

Thanks and cheers

A.Costa

**Comment by enghitalo** [comment]
> Hello,
> 
> V is an fantastic language!!:) I looked at the documentation and I'm wandering if there is the possibility to compile and execute V code within C code, like in TCC. 
> 
> Is there a convenient way to do this?
> 
> Thanks and cheers
> 
> A.Costa

Hi, A. Costa.

Yes it does. See https://github.com/vlang/v/blob/master/examples%2Fcall_v_from_c%2FREADME.md

**Reply by armincosta** [reply]
Thank you for the reply.

Yes I saw the documentation for calling v functions in C. Thanks

What I mean is to compile and execute embedded v code in C, like for example TCC does provide a function 'tcc_compile_string(..)' that compiles and executes the provided code directly

Is there way to do this in similar way?

Thanks a lot

Cheers,
Costa

**Reply by JalonSolov** [reply]
tcc doesn't know anything about V, so there's no way this could be done (unless you want to modify tcc to handle this).

It will be simpler to create shared libraries that either C or V code can call.

**Reply by armincosta** [reply]
Thank you for your reply. Sorry I think I did not express my answer correctly enough.

I was just taking TCC as example, which has such kind of functions (compile and execute at runtime)

My question is:
Does V have a function (or set of functions) that allow to directly compile and execute V code at rutime, from C , by simply passing the v code as a string? e.g 'v_compile_and_execute_string(my_v_code)'

I'm new to V so I may not fully understand the concepts yet.

Thanks for the patience

**Reply by JalonSolov** [reply]
No, V doesn't have a feature like that.

There may be a way around it using the hot-code reloading feature, but it would be tricky to set up correctly, and I don't know if anyone has tried it or not.

**Reply by armincosta** [reply]
Thank you very much for the suggestion.

Cheers
Costa

**Reply by Wajinn** [reply]
> tcc doesn't know anything about V

But there is [vtcc](https://github.com/felipensp/vtcc). Just mentioning it.

**Reply by armincosta** [reply]
Great!!! Thanks a lot for the hint
